### DISCUSSION TITLE
Tick labels format and auto range

### DISCUSSION THREAD

**Discussion by roman-59650** [GSI-CS-CO/chart-fx#657] [Q&A] [answered]
Hi, I have two questions:
1) How to format numbers in tick labels in order to show something like "50000.0" instead of "5.0000E4" ?
2) Is it possible to autorange y axis when one manually sets the range of x axis? so that maximum and minimum values on y axis would be the ones from the range of corresponding x values and not the minimum and maximum from the whole dataset. 
Thanks!

**Comment by wirew0rm** [comment, marked as answer]
Hi, welcome to chart-fx :wave: 

1) without actually trying this out, i would point you towards https://github.com/ennerf/chart-fx/blob/56a6dcb5e8e420845bde48f6f21a3a4498fb1240/chartfx-chart/src/main/java/io/fair_acc/chartfx/axes/spi/format/DefaultFormatter.java#L39 setting this to `Plain` should disable the exponential form

2) We thought about this for a long time and while for some cases this would be trivial to add it gets hard as soon as you have huge dynamic datasets there is no efficient algorithm to recompute the range and also the UX of which auto ranging axis depends on which other auto ranging axis is not easy to get right.
I would suggest to not use chartfx's auto ranging but implement your own using bindings from the x-Axis's range property to the y-Axis range, which recomputes the range. It might be good to have a sample for that since it is a recurring request. I'll try to find the time for it, in the meantime feel free to share your approach here.
Also one of the financial samples has an implementation for that, but there is a lot of other things around it, which make it difficult to just copy from there.

hope this as least helps to point you in the right direction.

**Reply by roman-59650** [reply]
Thanks for a quick answer! I basically found out myself the answer to the first question like 10 minutes after posting it, using a custom `StringConverter` in `setTickLabelFormatter` method.  Sorry 😅 for bothering. As for the second question, I still cannot find any range property that can be bound to neither in `DefaultNumericAxis` nor by calling `chart.getXAxis()`. What I found is `chart.getXAxis().autoRangingProperty()` which does not suit exactly my goals. I'll try to implement my own approach, no problem. Thanks again for a great library!

**Reply by wirew0rm** [reply]
Hey, glad that you were able to work it out and enjoy using chart-fx.
Feel free to share your solution and your custom implementation for dynamic autoranging here to also help other users with the same problems if it's something that can be easily shared as a small codesnipped or a link to your project :)
